### ISSUE TITLE
improve SEO in docs

### ISSUE THREAD

**Issue by erohmensing** [RasaHQ/rasa#6764]
- [ ] make sure every page has a `description` and that it is up to date (does not reference old product names)
- [ ] check header heirarchy -- each page should start with h2 (`##`) as the h1 is taken care of by the title. from there, headers underneath should be sequential with no skipping. 
- [ ] remove references to old product names (rasa core, rasa nlu) where possible

**Comment by lunelson** [comment]
@erohmensing a couple of additions:

- `description` right now is AFAIK only used for content of the `<meta name="description" content="...">` tag, and should be not much longer than a tweet: 140 characters is safe; max is considered to be between 150 and 170. It should add useful semantic information but not be stuffed with keywords (this will be penalized).
- `title` right now populates both the `h1` that you see on the page and the `<title>` tag in the head; this may not be ideal, since a very long title will look bad on the page but could be good for SEO...so we could consider adding an optional `metaTitle` property in the _frontmatter_ which falls back to `title` but would allow you to write something longer if you want. As for max length, [this preview tool suggests 78 characters](https://zeo.org/seo-tools/serp-preview/).

@sta-cee does that sound right? can you think of anything to add here?

**Comment by erohmensing** [comment]
>  It should add useful semantic information but not be stuffed with keywords (this will be penalized).

Very interesting, i think before this was specifically keyword stuffed!

**Comment by sta-cee** [comment]
@lunelson @erohmensing Best practices from our SEO agency:

**H1**
Each page have only one H1 header tag, placed high in the page’s content and populated with content relevant to the page’s content including a targeted key phrase. The H1 header should not match the Page title too closely

**Page titles**
Limit the page title to no more than 55 characters. Ideally, page titles will be no more than 45 characters. Note that Google displays only the first ~55 characters in their search results pages. Include target key phrases first, as this weighs the heaviest. Copy should be written with users in mind.

**Meta descriptions** 
Should be limited to 156 characters, include targeted key phrases, descriptive information about the page, a call-to-action, and be written with users in mind.

---

I don't really think we'll be able to incorporate target key phrases because our docs go into very specific topics but we should try to make sure the H1 and title are different if that's possible. 

I asked them for a recommendation on standardizing our doc page titles (especially since they gave us the feedback that a lot of our doc titles are too short). I was hoping for something like, "Rasa Documentation | Policies" but here's what they got back to me with:

Recommendation this for the page title "Contextual Assistant Policy Configuration | Rasa" (https://rasa.com/docs/rasa/core/policies/)

I don't think this is something we should necessarily do or try to force because that page title is a bit off. Unfortunately they mentioned that using "Rasa Documentation" to describe each docs page is not good in the sense that there will be many page titles with duplicate qualities.

**Comment by akelad** [comment]
@sta-cee is there an option of marketing helping out with this generally? I feel like y'all are probably more experienced in that aspect. We've removed this from the milestone for the final 2.0 release, because I don't think we'll get to it by then, and I don't think it's release critical

**Comment by lunelson** [comment]
I'm going to push support for a `metaTitle` frontmatter property this week, this will make it possible to have `h1` be different from `document.title`

**Comment by sta-cee** [comment]
yeah I think that makes sense @akelad - looping in @karen-white so we can add to our list for Q4 OKR planning

**Comment by lunelson** [comment]
`metaTitle` and `metaDescription` are now supported in frontmatter of MDX files, in the `v0.8.0` release of the docusaurus theme

**Comment by fullpoint** [comment]
@sta-cee @akelad I'm good with marketing helping with this, but not with us taking complete ownership of it. Experience is 10% of doing this, it's 90% sweat, and we should share that. Maybe Stacee could create guidelines and a list of all pages that need this work, and we could all divide and conquer?

**Comment by sta-cee** [comment]
@fullpoint sound good, I'll capture guidelines on a notion page and @lunelson already has it on his [list](https://app.asana.com/0/1189355535659914/1193214334325100/f) to export titles and descriptions this week

**Comment by akelad** [comment]
yeah that could work! We need to address the remaining pages to be re-written according to the styleguide anyways, so we could do the SEO thing as part of that
